node_modules/oae-principals/lib/api.user.js

```diff
@@ -723,3 +961,109 @@ var _setAdmin = function(ctx, adminType, isAdmin, principalId, callback) {
         return PrincipalsDAO.setAdmin(adminType, isAdmin, principalId, callback);
     });
 };
+
+/**
+ * Send an email token to a user
+ *
+ * @param  {Context}    ctx             Standard context object containing the current user and the current tenant
+ * @param  {User}       user            The user to send the email token to
+ * @param  {Function}   callback        Standard callback function
+ * @param  {Object}     callback.err    An error that occurred, if any
+ */
+var sendEmailToken = module.exports.sendEmailToken = function(ctx, user, callback) {
+    callback = callback || function(err) {
+        if (err) {
+            log().error({'err': err, 'userId': user.id}, 'Unable to send a user an email verification email');
+        }
+    };
+
+    // Generate a token that will be sent to the user's email address
+    var token = _generateEmailToken();
+
+    // Store the token
+    PrincipalsDAO.storeEmailToken(token, user.id, function(err) {
+        if (err) {
+            return callback(err);
+        }
+
+        // Send an email to the specified e-mail address
+        var data = {
+            'tenant': ctx.tenant(),
+            'user': user,
+            'baseUrl': TenantsUtil.getBaseUrl(ctx.tenant()),
+            'skinVariables': require('oae-ui').getTenantSkinVariables(ctx.tenant().alias),
+            'token': token
+        };
+        EmailAPI.sendEmail('oae-principals', 'verify', user, data, {'hash': token, 'ignoreVerification': true}, callback);
+    });
+};
+
+/**
+ * Resend an email token
```

Comment:
This was meant to be used for when an email got lost or when you didn't respond within 24 hours. I just noticed that I'm re-using the email tokens for guest accounts. Regular users (who log in through SSO) could exploit this to add a local loginId to their own account, I'll have to put something in place to stop that.